src/ethereum_spec_tools/evm_tools/fixture_loader.py

```diff
@@ -346,55 +360,57 @@ def json_to_withdrawals(self, raw: Any) -> Any:
 
         return self._module("fork_types").Withdrawal(*parameters)
 
-    def json_to_blocks(
+    def json_to_block(
         self,
-        json_blocks: Any,
-    ) -> Tuple[List[Any], List[Hash32], List[bytes]]:
+        json_block: Any,
+    ) -> Tuple[Any, Hash32, bytes, Optional[str]]:
         """Converts json block data to a block object"""
-        blocks = []
-        block_header_hashes = []
-        block_rlps = []
-
-        for json_block in json_blocks:
-            if "rlp" in json_block:
-                # Always decode from rlp
-                block_rlp = hex_to_bytes(json_block["rlp"])
-                block = rlp.decode_to(self.Block, block_rlp)
-                blocks.append(block)
-                block_header_hashes.append(rlp.rlp_hash(block.header))
-                block_rlps.append(block_rlp)
-                continue
-
-            header = self.json_to_header(json_block["blockHeader"])
-            transactions = tuple(
-                self.json_to_tx(tx) for tx in json_block["transactions"]
-            )
-            uncles = tuple(
-                self.json_to_header(uncle)
-                for uncle in json_block["uncleHeaders"]
-            )
-
-            parameters = [
-                header,
-                transactions,
-                uncles,
-            ]
+        block_exception = None
+        for key, value in json_block.items():
+            if key.startswith("expectException"):
+                block_exception = value
+                break
 
-            if "withdrawals" in json_block:
-                withdrawals = tuple(
-                    self.json_to_withdrawals(wd)
-                    for wd in json_block["withdrawals"]
-                )
-                parameters.append(withdrawals)
+        if "rlp" in json_block:
+            # Always decode from rlp
+            block_rlp = hex_to_bytes(json_block["rlp"])
+            try:
+                block = rlp.decode_to(self.Block, block_rlp)
```

Comment:
Do we fail the test if an expected exception doesn't occur?

Reply:
There are 2 cases here

1. Some other exception is raised which is not an `ExpectedRLPException`. In this case, the same exception will be raised and the test will fail. We only catch the `ExpectedRLPException` [here](https://github.com/gurukamath/execution-specs/blob/2f2da0c601e9870a52e41e25aacfe751de699207/tests/helpers/load_state_tests.py#L76)
2. No exception is raised. In this case, the further processing will continue. However, one of the later checks will have to fail. Assertion of state_root, block_hash etc. for example.

Reply:
And it's guaranteed that one will fail?

Reply:
Yes. When we successfully process an invalid block, the last_block_hash will not match. The post state will also not match in most cases.

However, in the most recent upsate I pushed, I check explicitly for the this with `with pytest.raises(RLPDecodingError):`. So we don't have to rely on the above checks failing.